oauth2/src/main/java/com/okta/spring/boot/oauth/OktaOAuth2Configurer.java

```diff
@@ -75,6 +79,20 @@ public void init(HttpSecurity http) throws Exception {
                 } else {
                     log.debug("OAuth resource server not configured due to missing issuer property");
                 }
+
+                if (oktaOAuth2Properties.isOpaque() && !isEmpty(resourceServerProperties.getOpaquetoken().getIntrospectionUri())) {
+                    OpaqueTokenIntrospector introspectionClient = new NimbusOpaqueTokenIntrospector(
+                        resourceServerProperties.getOpaquetoken().getIntrospectionUri(),
+                        oktaOAuth2Properties.getClientId(),
+                        oktaOAuth2Properties.getClientSecret());
+
+                    OpaqueTokenAuthenticationProvider opaqueTokenAuthenticationProvider =
+                        new OpaqueTokenAuthenticationProvider(introspectionClient);
+
+                    http.oauth2ResourceServer().opaqueToken()
+                        .authenticationManager(authentication -> opaqueTokenAuthenticationProvider
+                            .authenticate(SecurityContextHolder.getContext().getAuthentication()));
```

Comment:
We would also need to deal with the group claim converter similar to line 98/116 below

Reply:
https://github.com/spring-projects/spring-security/blob/master/oauth2/oauth2-resource-server/src/main/java/org/springframework/security/oauth2/server/resource/introspection/NimbusOpaqueTokenIntrospector.java#L182-L217 does exactly this. 

We can make use of this introspector:

```
OpaqueTokenIntrospector opaqueTokenIntrospector = context.getBean(OpaqueTokenIntrospector.class);
http.oauth2ResourceServer().opaqueToken().introspector(opaqueTokenIntrospector);
```

Reply:
addressed this.